### PR TITLE
Finished most workshop tasks still need to finish the last one with t…

### PR DESCRIPTION
**Pull request by Wmd2452**
…he blending

### CODE REVIEW

**Comment by st20231722** [comment]
Merged branches created on accidnet

File: glDemo/main.cpp [hunk 1]
```diff
@@ -111,21 +160,106 @@ int main() {
...
+	glColor3f(0.0f, 0.0f, 1.0f);
+	glVertex2f(0.5f, -0.5f);
+
+	glColor3f(0.0f, 1.0f, 1.0f);
+	glVertex2f(0.5f, 0.5f);
+
+	glEnd();
+}
+
+void randomPoints()
```
Comment (pna21842) [inline glDemo/main.cpp:242]: Your random points function should merge the 2 for loops.  At the moment you iterate through all 100 points setting the size and colour.  Then you iterate through all the points and draw them.  However, they'll all be drawn with the size and colour that was set in the last iteration of the previous loop.

The structure should be

for i = 0..100
{
 Set colour
 Set size
 glBegin points
   Draw single point
 glEnd
}

File: glDemo/main.cpp [hunk 2]
```diff
@@ -86,6 +100,41 @@ int main() {
...
 	gluOrtho2D(-1.1f, 1.1f, -1.1f, 1.1f);
 
+
+	random_device rd;
+	engine = mt19937(rd());
+	range = uniform_real_distribution<float>(-1.0f, 1.0f);
+
+	random_device dr;
+	engine2 = mt19937(dr());
+	range2 = uniform_real_distribution<float>(-1.0f, 20.0f);
```
Comment (pna21842) [inline glDemo/main.cpp:110]: You set the range for the point size from -1 to 20.  This ideally should be +1 to 20.

File: glDemo/main.cpp [hunk 3]
```diff
@@ -86,6 +100,41 @@ int main() {
...
+
+	// Iterate 100 times (once for each vertex) and calculate and store the random x and y values
+	for (int i = 0; i < 100; i++) {
+
+		float x = range(engine);
+		float y = range(engine);
+
+		vertexCoords[i] = glm::vec2(x, y);
+
+		float r = range(engine);
```
Comment (pna21842) [inline glDemo/main.cpp:126]: In the colour setup code you use the same range as the position data, but this goes from -1 to +1.  Ideally for the colour data you'd have a range from 0 to 1

File: glDemo/main.cpp [hunk 4]
```diff
@@ -86,6 +100,41 @@ int main() {
 	// *** setup viewplane to the appropriate size
 	gluOrtho2D(-1.1f, 1.1f, -1.1f, 1.1f);
 
+
+	random_device rd;
+	engine = mt19937(rd());
+	range = uniform_real_distribution<float>(-1.0f, 1.0f);
+
+	random_device dr;
+	engine2 = mt19937(dr());
```
Comment (pna21842) [inline glDemo/main.cpp:109]: Don't need 2nd engine - can use single engine to drive multiple ranges / distributions

**Pull request closed without merging**